### PR TITLE
Store NefFile in ContractState

### PR DESCRIPTION
**Pull request by erikzhang**

### CODE REVIEW

**Comment by erikzhang** [comment]
Go?

File: src/neo/SmartContract/Native/NativeContract.cs [hunk 1]
```diff
@@ -27,21 +27,31 @@ public abstract class NativeContract
...
                 sb.EmitPush(Name);
                 sb.EmitSysCall(ApplicationEngine.System_Contract_CallNative);
-                this.Script = sb.ToArray();
+                script = sb.ToArray();
             }
-            this.Hash = Helper.GetContractHash(UInt160.Zero, Script);
+            this.Nef = new NefFile
+            {
+                Compiler = nameof(ScriptBuilder),
+                Version = typeof(ScriptBuilder).Assembly.GetVersion(),
```
Comment (shargon) [inline src/neo/SmartContract/Native/NativeContract.cs:49]: We need a deterministic value

Comment (shargon) [inline src/neo/SmartContract/Native/NativeContract.cs:49]:
```suggestion
                Version = "3.0.0.0",
```

Comment (erikzhang) [reply src/neo/SmartContract/Native/NativeContract.cs]: Fixed.

**Review by shargon — APPROVED** [review]

**Pull request merged**